.github/workflows/node.js.yml

```diff
@@ -0,0 +1,107 @@
+name: Node.js CI
+
+on:
+  push:
+    branches:
+      - jp-gha-conversion
+      - staging
+      # - master
+  pull_request:
+    branches:
+      - jp-gha-conversion
+      - staging
+      # - master
+  workflow_dispatch:
+
+jobs:
+  PreDeployTesting:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Repository checkout
+        uses: actions/checkout@v3
+
+      - name: Node.js setup
+        uses: actions/setup-node@v3
+        with:
+          node-version: '16.x'
+
+      - name: 📦 Install dependencies
+        run: yarn install
+
+      - name: 💅 Prettier check
+        run: yarn prettier:check
+
+      - name: 💨 Lint check
+        run: yarn lint
+
+      - name: 🏗️ Build app
+        run: yarn ng build --prod
+
+  Deploy:
+    if: ${{ github.event_name == 'push' && (github.ref == 'refs/heads/master' || github.ref == 'refs/heads/staging')}}
+    runs-on: ubuntu-latest
+    environment:
+      name: ${{ (github.ref == 'refs/heads/master' && 'prod') || 'stage' }}
+    needs: PreDeployTesting
+    permissions:
+      id-token: write
+      contents: read
+
+    steps:
+      - name: Repository checkout
+        uses: actions/checkout@v3
+
+      - name: Install AWSCLI
+        run: pip install --user awscli
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@036a4a1ddf2c0e7a782dca6e083c6c53e5d90321
+        with:
+          aws-region: us-east-1
+          role-to-assume: ${{ secrets.AWS_IAM_ROLE_ARN }}
+
+      - name: Node.js setup
+        uses: actions/setup-node@v3
+        with:
+          node-version: '16.x'
+
+      - name: 📦 Install dependencies
+        run: yarn install
+
+      - name: 🏗️ Build app
+        run: yarn ng build --prod
+
+      - name: Copy embed.js
+        run: cp embed/embed.js dist/embed.js
+
+      - name: Copy example.html
+        run: cp embed/example.html dist/example.html
+
+      - name: Copy mobile directory
+        run: cp -r mobile/ dist/mobile/
+
+      - name: Update embed host (staging)
+        if: github.ref == 'refs/heads/staging'
+        run: sed -i "s%$appDomainPlaceholder%$appDomain%g" dist/embed.js
+        env:
+          appDomainPlaceholder: '{appDomain}'
+          appDomain: 'https://stage.knowgod.com'
```

Comment:
this should be configured in either the GHA environment or terraform (probably terraform).

Reply:
@frett I caught that too and was already planning to use `${{ secrets.AWS_S3_BUCKET }}` but how would I add `https://` to the beginning? Would `https://${{ secrets.AWS_S3_BUCKET }}` work?

Reply:
don't use `AWS_S3_BUCKET` as the secret, there is no guarantee that the bucket name will match the hostname for the actual site

Reply:
But isn't that what I'm doing (with the code you gave me) in the "Deploy to s3" step? I double checked the values assigned in Terraform and they're `staging.knowgod.com` and `knowgod.com`.

Side note: I had to add "s3://" to each line of `run` to get that step to work.

Reply:
that is for the next step, what this step is doing is updating the `embed.js` to put in the web url you are serving the site from, which is not always going to be the same thing as the s3 bucket name

Reply:
this step actually has nothing to do with AWS or S3, it just is preparing `embed.js` to be used on the web

Reply:
I get that the steps aren't directly related, but back to my first question, how should I concatenate the `https://` to the front of `${{ secrets.HOST_NAME }}`?

Reply:
`https://${{ secrets.HOST_NAME }}`